### PR TITLE
Initialize name and current_turn when a game is created

### PR DESCRIPTION
**Pull request by kellylougheed**
Updated create action to initialize current_turn as 0 and the game name as "Game #{id}." Adjusted tests for the create action to rely on the game count rather than the auto-generated name.

Once we integrate devise and find a way to intake the opposing player, we can also update the create action randomly assign the players black and white.

### CODE REVIEW

**Comment by PamBWillenz** [comment]
Looks good, Kelly. Wondering if you can use just one `@game.save` by putting it at the end of that method?

**Comment by kellylougheed** [comment]
Pam, unfortunately I could only get the @game.id (for the @game.name) if I saved the game first -- so I ended up saving it twice. I agree it's not ideal!

**Comment by lasley** [comment]
Could always drop the `name` column entirely and provide an abstract attribute to the model providing dynamic name computation:

```
attr_accessor :name
def name
  "Game #{self.id}"
end
```

**Comment by kellylougheed** [comment]
Dave, that might be cleaner -- thanks for the idea. The only thing I've been using the name for is to test the "update" action, since we decided that "name" would be more innocuous to have in the params than "current_turn." But I could test with "current_turn" as well if I added it to the params.

**Comment by lasley** [comment]
No lost functionality, just stored in database or computed on the fly. IMO there's no reason to store computed data unless the computation is heavy.

Yeah `current_turn` is probably the safest to expose. 😆  at the ping-pong of the attribute exposing

**Comment by lasley** [comment]
Ok actually lost functionality is no searching in the where query, which is typically also irrelevant for computed because you just search the properties it's computed from instead.

**Comment by kellylougheed** [comment]
OK, we can have the name be an attribute. I changed the exposed param to current_turn and updated the tests.

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -25,6 +31,8 @@ def edit
   # POST /games.json
   def create
     @game = Game.new(game_params)
+    @game.current_turn = 0
+    @game.save
```
Comment (lasley) [inline app/controllers/games_controller.rb]: You can remove this in favor of the `@game.save` on L38

Comment (kellylougheed) [inline app/controllers/games_controller.rb]: Done!

**Comment by lasley** [comment]
Thanks @kellylougheed

**Pull request merged**